### PR TITLE
TST: Test clone and get on public Gin repo

### PR DESCRIPTION
**Pull request by adswa**
Here is a test for cloning a Gin dataset and getting content (https://github.com/datalad/datalad/issues/2680#issuecomment-614781965). ping @mih 

I'm not sure against which branch this needs to be PR'ed. Let me know if I should rebase.

### CODE REVIEW

File: datalad/core/distributed/tests/test_clone.py [hunk 1]
```diff
@@ -910,3 +910,19 @@ def test_clone_unborn_head_sub(path):
...
+
+@skip_if_no_network
+@with_tempfile
+def test_gin_cloning(path):
+    # can we clone a public ds anoynmously from gin and retrieve content
+    ds = clone('https://gin.g-node.org/datalad/datalad-ci-target', path)
+    ok_(ds.is_installed())
+    ok_(ds.repo.file_has_content('annex/two') is False)
+    ok_(ds.repo.is_under_annex('git/one') is False)
+    result = ds.get("annex/two")
```
Comment (mih) [inline datalad/core/distributed/tests/test_clone.py:923]: I suspect that this will fail on windows, due to the hard-coded `/`

File: datalad/core/distributed/tests/test_clone.py [hunk 2]
```diff
@@ -910,3 +910,19 @@ def test_clone_unborn_head_sub(path):
...
     eq_(ds_cloned_sub.repo.get_active_branch(), "other")
+
+
+@skip_if_no_network
+@with_tempfile
+def test_gin_cloning(path):
+    # can we clone a public ds anoynmously from gin and retrieve content
+    ds = clone('https://gin.g-node.org/datalad/datalad-ci-target', path)
+    ok_(ds.is_installed())
+    ok_(ds.repo.file_has_content('annex/two') is False)
```
Comment (mih) [inline datalad/core/distributed/tests/test_clone.py:921]: Instead of `ok_(... is False)` I'd personally prefer to use `eq_(..., False)`, but there is also `nok_()`.

File: datalad/core/distributed/tests/test_clone.py [hunk 3]
```diff
@@ -910,3 +910,19 @@ def test_clone_unborn_head_sub(path):
...
+    ok_(ds.repo.is_under_annex('git/one') is False)
+    result = ds.get("annex/two")
+    assert_result_count(result, 1)
+    assert_status('ok', result)
+    eq_(result[0]['path'], op.join(ds.path, 'annex/two'))
+    eq_(result[0]['annexkey'], ds.repo.get_file_key('annex/two'))
```
Comment (mih) [inline datalad/core/distributed/tests/test_clone.py:927]: This particular one seems out of scope for the purpose of this test. It checks whether the internal reporting is consistent. Instead, I would check if the content of `git/one` is `one` and that of `annex/two` is `two` (both with `ok_file_has_content()`). I think that would be more goal oriented (we get from GIN what we put there).

File: datalad/core/distributed/tests/test_clone.py (hunk 1 above)
Comment (adswa) [reply datalad/core/distributed/tests/test_clone.py]: I now use `op.join()`

File: datalad/core/distributed/tests/test_clone.py (hunk 2 above)
Comment (adswa) [reply datalad/core/distributed/tests/test_clone.py]: done

File: datalad/core/distributed/tests/test_clone.py (hunk 3 above)
Comment (adswa) [reply datalad/core/distributed/tests/test_clone.py]: thanks for the suggestion, done.

**Comment by adswa** [comment]
Thanks @mih! I have pushed changes that hopefully address your comments.

**Comment by adswa** [comment]
There is one failing test from #4370 that I believe is unrelated to this change, and if I understand https://github.com/datalad/datalad/pull/4405#issuecomment-614770310 correctly, its because this PR is based on master:

```

======================================================================
ERROR: datalad.core.distributed.tests.test_clone.test_clone_unborn_head
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/tests/utils.py", line 691, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/core/distributed/tests/test_clone.py", line 875, in test_clone_unborn_head
    ds_origin.repo.call_git(["checkout", "-b", "chooseme", "abc~2"])
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/support/gitrepo.py", line 2107, in call_git
    check_fake_dates=True)
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/support/gitrepo.py", line 326, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/support/gitrepo.py", line 2058, in _git_custom_command
    expect_fail=expect_fail)
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/cmd.py", line 143, in run_gitcommand_on_file_list_chunks
    results.append(func(cmd, *args, **kwargs))
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/cmd.py", line 1114, in run
    *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.5.9/x64/lib/python3.5/site-packages/datalad/cmd.py", line 930, in run
    raise exc
datalad.support.exceptions.CommandError: CommandError: 'git checkout -b chooseme 'abc~2'' failed with exitcode 128 under /crippledfs/datalad_temp_test_clone_unborn_headaqr3x21d/a [err: 'fatal: 'abc~2' is not a commit and a branch 'chooseme' cannot be created from it']
Obscure filename: str=b' ;abc\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82' repr=' ;abcΔЙקم๗あ'
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: GIT_PYTHON_GIT_EXECUTABLE='/usr/lib/git-annex.linux/git' PATH='/opt/hostedtoolcache/Python/3.5.9/x64/bin:/opt/hostedtoolcache/Python/3.5.9/x64:/usr/share/rust/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/home/runner/.dotnet/tools:/home/runner/.config/composer/vendor/bin' LANG='C.UTF-8'
```

**Comment by bpoldrack** [comment]
>   There is one failing test from #4370 that I believe is unrelated to this change, and if I understand #4405 (comment) correctly, its because this PR is based on master:

Correct.

**Comment by mih** [comment]
Which they have. Test failure is unrelated.
